sat/cli/bootsys/mgmt_power.py

```diff
@@ -260,6 +261,54 @@ def finish_shutdown(hosts, username, password, ncn_shutdown_timeout, ipmi_timeou
             sys.exit(0)
 
 
+def set_next_boot_device_to_disk(ssh_client, ncns):
+    """
+    Set the next boot device to a disk for a list of nodes via SSH.
+
+    This function connects to each node via SSH, retrieves the current boot order, identifies the first available disk,
+    and sets the next boot device to the specified disk.
+
+    Args:
+        ssh_client (paramiko.SSHClient): An active SSH client used to connect to the nodes.
+        ncns (list): A list of node names (NCNs) to configure.
+    """
+    command = "efibootmgr"
+
+    for ncn in ncns:
+        try:
+            ssh_client.connect(ncn)
+        except (SSHException, socket.error) as err:
+            LOGGER.warning(f'Unable to connect to node {ncn}: {err}')
+            continue
+
+        try:
+            _, stdout, stderr = ssh_client.exec_command(command)
+            exit_code = stdout.channel.recv_exit_status()
+            if exit_code == 0:
```

Comment:
I think we should log a warning if the `exit_code` is non-zero. So maybe you want to invert this condition, log a warning and continue, and then put the other code either in an else, or outside of the `if` just below. Something like:

```
            if exit_code != 0:
                LOGGER.warning(f'Unable to determine boot order of {ncn}, {command} exited with exit code {exit_code}')
                continue
            # Then, other code when efibootmgr was successful goes here
```